### PR TITLE
Feature: Location on Articles and Users

### PR DESCRIPTION
**Pull request by aatifjiwani**
Added gem and helper methods to handle IPStack API responses and make API requests
Updated show view for Article and User to implement location
Updated create view for Article and User to allow for users to choose if they want location attached
Updated create action for Article and User to handle location if applicable
Added location feature specs.

### CODE REVIEW

**Comment by aatifjiwani** [comment]
### Show article view that displays location when posted 
screen shot 2018-06-27 at 2 59 43 pm

<br>
<br>

### Show user view that displays location when posted 
screen shot 2018-06-27 at 2 59 32 pm

File: Gemfile [hunk 1]
```diff
@@ -71,6 +75,7 @@ group :test do
   gem 'selenium-webdriver'
   # Easy installation and use of chromedriver to run system tests with Chrome
   gem 'chromedriver-helper'
+  gem 'vcr', '~> 4.0'
```
Comment (efatsi) [inline Gemfile]: doesn't look like you're making use of this gem. would be worth getting it configured and wrapped around the new location specs

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -37,7 +38,12 @@ def index
   def create
     @current_user = current_user
     @article = @current_user.articles.new(article_params)
-
+    
+    if params[:article][:location]
+      api_request = HTTP.get(location_api_url).as_json
+      @article.location = get_only_location(JSON.parse(api_request["body"][0]))
```
Comment (efatsi) [inline app/controllers/articles_controller.rb]: Having the location API methods live in a view helper is perhaps not the best place. Code in `/app/helpers` should really just be focused on making your life easier in the view layer.

I'd recommend moving all of this functionality into a new class within the `/app/lib` directory. Would be cool if the class took in an IP address on initialize and had a single method to spit out the location data. So your code here could look like: `@article.location = Geolocator.new(request.remote_ip).location` (name the things whatever you think is best), and the class itself would handle all of the nitty gritty around API interactions and slicing out the desired data.

Benefit of something like that would be that all the location code would be encapsulated in one place, you could minimize the impact this has on your controllers, and you could unit test the geolocating class if you felt inclined.

File: app/views/articles/new.html.erb [hunk 3]
```diff
@@ -32,6 +32,12 @@
             createFormattingToolBar();
           </script> 
         </p>
+      
+        <p class="form-p">
+          <%= form.check_box :location, {}, 1, nil %>
+          <%= form.label :location, 'Add Current Location to Article', class: "checkbox", style: "display: inline;" %>
```
Comment (efatsi) [inline app/views/articles/new.html.erb]: a little strange that you're using the `:location` attribute to pass in "I want this thing to be geolocated". Muddies up the nature of the `location` attribute a bit. Another option could be to set `attr_accessor :should_geolocate` on the model (which would make `should_geolocate` callable on an Article instance, but not persist the value to the db) and use that here instead. No strong opinions here, so whatever you think is cleanest works for me.

File: app/views/shared/_location.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% if object.location %>
+  <p class="location">
+    <i class="fas fa-map-marker-alt"></i> 
+    <%= object.location["city"] %>, <%= object.location["country_name"] %>
+  </p>
+<% end %>
```
Comment (efatsi) [inline app/views/shared/_location.html.erb]: awww, no map?

File: Gemfile (hunk 1 above)
Comment (aatifjiwani) [reply Gemfile]: Configured

File: app/views/shared/_location.html.erb (hunk 4 above)
Comment (aatifjiwani) [reply app/views/shared/_location.html.erb]: Added a new class to render static map images from the Google Map API

**Comment by aatifjiwani** [comment]
### How the new user profile looks with image

<br>
screen shot 2018-06-28 at 11 03 22 am

**Pull request merged**